src/js/program/media-element-pool.ts

```diff
@@ -20,30 +36,31 @@ export default function MediaElementPool() {
     let primed = false;
 
     return {
-        primed() {
+        primed(): boolean {
             return primed;
         },
-        prime() {
+        prime(): void {
             elements.forEach(primeMediaElementForPlayback);
             primed = true;
         },
-        played() {
+        played(): void {
             primed = true;
         },
-        getPrimedElement() {
-            if (pool.length) {
+        getPrimedElement(): HTMLVideoElement | null {
+            const el = pool.shift();
+            if (el) {
                 // Shift over pop so that we cycle through elements instead of reusing the same one
-                return pool.shift();
+                return el;
             }
             return null;
         },
-        getAdElement() {
+        getAdElement(): OptionalUndefVideo {
```

Comment:
I think adElement is always defined.

Reply:
same problem with `pool.shift` the possibility of it being undefined causes a need for the optional undef. we should think more holistically about this, i think we may uncover a smarter approach when dependencies that call this function are type

Reply:
I thought `adElement` was a fixed value. It's assigned using `shift()`?
Either break out the media element creating function to assign `adElement` or assign `adElement as HTMLMediaElemt` because we know better (we're always filling the pool and shifting one element out for ads).

Reply:
updating!